install.sh

```
@@ -0,0 +1,230 @@
+}
+
+log_warning() {
+    echo -e "${YELLOW}⚠️  $1${NC}"
+}
+
+log_error() {
+    echo -e "${RED}❌ $1${NC}"
+}
+
+detect_platform() {
+    local os=$(uname -s | tr '[:upper:]' '[:lower:]')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)